src/ops.c

```diff
@@ -249,18 +256,34 @@ shift_line(
 	}
 	else
 	    i += amount;
-	count = (long long)i * (long long)sw_val;
+	count = i * sw_val;
     }
     else		// original vi indent
     {
 	if (left)
 	{
+#ifdef HAVE_NO_LONG_LONG
+	    count -= (long)sw_val * (long)amount;
+#else
 	    count -= (long long)sw_val * (long long)amount;
+#endif
 	    if (count < 0)
 		count = 0;
 	}
 	else
-	    count += (long long)sw_val * (long long)amount;
+	{
+#ifdef HAVE_NO_LONG_LONG
+	    if ((long)sw_val * (long)amount > INT_MAX - count)
+		count = INT_MAX;
+	    else
+		count += (long)sw_val * (long)amount;
+#else
+	    if ((long long)sw_val * (long long)amount > INT_MAX - count)
+		count = INT_MAX;
+	    else
+		count += (long long)sw_val * (long long)amount;
+#endif
```

Comment:
i'm curious to ask: 
the port here the trick is like this to make it pick up different length 'count'?
vs/but not a way to mock it work like a fake 64bit long?
not sure if it's possible, or con/pro or outage.